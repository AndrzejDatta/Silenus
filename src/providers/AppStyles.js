import { ThemeProvider } from 'styled-components';
import { theme } from 'assets/styles/theme';
import { GlobalStyle } from 'assets/styles/GlobalStyles';
export const AppStyles = ({ children }) => {
  return (
    <ThemeProvider theme={theme}>
      <GlobalStyle />
      {children}
    </ThemeProvider>
  );
};
