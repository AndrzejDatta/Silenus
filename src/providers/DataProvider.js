import { createContext, useState } from "react";
import { dataPL } from "assets/data/data";

export const dataContext = createContext({
  startPage: {},
  loginPage: {},
  Dashboard: {},
  MojeDoniczki: {},
  DodajDoniczke: {},
  MojeRośliny: {},
  KalendarzOpieki: {},
  KontoUżytkownika: {},
  ZidentyfikujRośline: {},
  PotDetails: {},
  PlantDetails: {},
  Navbar: {},
  images: {},
  navIcons: {},
  potIcons: {},
  categoryIcons: {},
  navIconsColor: [],
  setNavIconColor: () => {},
  changeColor: () => {},
  importAll: () => {},
});

const objectName = [
  "startPage",
  "loginPage",
  "Dashboard",
  "MojeDoniczki",
  "DodajDoniczke",
  "MojeRośliny",
  "KalendarzOpieki",
  "KontoUżytkownika",
  "ZidentyfikujRośline",
  "PotDetails",
  "PlantDetails",
  "Navbar",
];

export const DataProvider = ({ children }) => {
  for (let i = 0; i < dataPL.length; i++) {
    objectName[i] = { ...dataPL[i] };
  }
  const [
    startPage,
    loginPage,
    Dashboard,
    MojeDoniczki,
    DodajDoniczke,
    MojeRośliny,
    KalendarzOpieki,
    KontoUżytkownika,
    ZidentyfikujRośline,
    PotDetails,
    PlantDetails,
    Navbar,
  ] = objectName;

  const [navIconsColor, setNavIconColor] = useState([
    {
      name: "plant",
      src: "potted_plant.svg",
      isClicked: false,
    },
    {
      name: "calendar",
      src: "calendar.svg",
      isClicked: false,
    },
    {
      name: "camera",
      src: "camera.svg",
      isClicked: false,
    },
    {
      name: "user",
      src: "accountCircle.svg",
      isClicked: false,
    },
  ]);

  function importAll(r) {
    let images = {};
    r.keys().map((item, index) => {
      images[item.replace("./", "")] = r(item);
    });
    return images;
  }

  const images = importAll(
    require.context(
      "assets/icons/userAccountOption",
      false,
      /\.(png|jpe?g|svg)$/
    )
  );
  const navIcons = importAll(
    require.context("assets/icons/navIcons", false, /\.(png|jpe?g|svg)$/)
  );

  const potIcons = importAll(
    require.context("assets/icons/potStatus", false, /\.(png|jpe?g|svg)$/)
  );

  const categoryIcons = importAll(
    require.context("assets/icons/categoryIcons", false, /\.(png|jpe?g|svg)$/)
  );

  function changeColor(nameElement) {
    setNavIconColor((prevData) =>
      prevData.map((item) => {
        if (item.name === nameElement && !item.isClicked) {
          return {
            ...item,
            isClicked: true,
          };
        } else {
          return {
            ...item,
            isClicked: false,
          };
        }
      })
    );
  }

  return (
    <dataContext.Provider
      value={{
        startPage,
        loginPage,
        Dashboard,
        DodajDoniczke,
        MojeDoniczki,
        MojeRośliny,
        KalendarzOpieki,
        KontoUżytkownika,
        ZidentyfikujRośline,
        PotDetails,
        PlantDetails,
        Navbar,
        images,
        navIcons,
        potIcons,
        categoryIcons,
        navIconsColor,
        setNavIconColor,
        changeColor,
        importAll,
      }}
    >
      {children}
    </dataContext.Provider>
  );
};

//TODO:improve way
