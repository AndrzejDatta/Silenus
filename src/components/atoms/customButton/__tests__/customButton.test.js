import "jest-styled-components";
import renderer from "react-test-renderer";
import React from "react";
import CustomButton from "components/atoms/customButton/customButton.component";
import { render, screen } from "@testing-library/react";
import { AppStyles } from "providers/AppStyles";

describe("Test CustomButton component", () => {
  it("test default style", () => {
    const tree = renderer
      .create(
        <AppStyles>
          <CustomButton text="Text" />
        </AppStyles>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
    expect(tree).toHaveStyleRule("background-color", "green");
  });
  it("check text in button after rendere", () => {
    render(
      <AppStyles>
        <CustomButton text="test" fontSize="60px" />
      </AppStyles>
    );
    expect(screen.getByRole("button", { name: /test/i })).toHaveStyleRule(
      "font-size",
      "60px"
    );
  });
});
