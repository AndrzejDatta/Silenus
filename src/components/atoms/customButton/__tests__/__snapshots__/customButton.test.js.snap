// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test CustomButton component test default style 1`] = `
.c0 {
  width: 200px;
  height: 45px;
  font-size: 14px;
  font-family: "Poppins",sans-serif;
  background-color: green;
  color: black;
  border: 1px solid #EEF7F4;
  border-radius: 10px;
  margin: 0;
  padding: 0;
  cursor: pointer;
}

<button
  className="c0"
  color="black"
  fontSize="14px"
  onClick={[Function]}
  size="small"
  type="text"
>
  Text
</button>
`;
